src/sql/src/plan/lowering.rs

```diff
@@ -1032,6 +1032,224 @@ impl HirScalarExpr {
                                                 },
                                             );
 
+                                            let agg_col = record_col + 1 + input_arity;
+                                            reduce.project(
+                                                (record_col + 1..agg_col + 1).collect_vec(),
+                                            )
+                                        })
+                                    });
+                            SS::Column(inner.arity() - 1)
+                        }
+                        WindowExprType::Value(func) => {
```

Comment:
hic sunt dracones.

The core idea is similar to how scalar window functions are handled, the main changes are in how arguments to value window functions are encoded and passed around. I've took the opportunity to make the serialization of the args here a bit easier to handle from inside the aggregation function.

Reply:
These comments were super helpful. Makes me feel like I almost understand what's going on :D